### PR TITLE
[explorer] Implements Address Results Redesign

### PR DESCRIPTION
**Pull request by Andrew47**
This implements a redesign of the Address Results Page as detailed in the Figma Diagrams. This led to changes to the Owned Objects and Pagination Wrapper components. There is also a minor change to Object Results to re-instate centering of the content. The tests have also been updated.

Screenshot with live API:
image

Screenshot of pagination in mobile view:
image

### CODE REVIEW

File: explorer/client/src/components/ownedobjects/OwnedObjects.tsx [hunk 1]
```diff
@@ -217,118 +234,162 @@ function GroupView({ results }: { results: resultType }) {
...
+                                                                    )}
+                                                                    category="objects"
+                                                                    isCopyButton={
+                                                                        false
+                                                                    }
+                                                                    alttext={truncate(
+                                                                        String(
+                                                                            value
+                                                                        ),
+                                                                        19
```
Comment (666lcz) [inline explorer/client/src/components/ownedobjects/OwnedObjects.tsx:366]: ditto

File: explorer/client/src/components/ownedobjects/OwnedObjects.tsx [hunk 2]
```diff
@@ -217,118 +234,162 @@ function GroupView({ results }: { results: resultType }) {
...
+                                                    );
+                                                } else {
+                                                    return (
+                                                        <Longtext
+                                                            text={String(value)}
+                                                            category="objects"
+                                                            isCopyButton={false}
+                                                            alttext={truncate(
+                                                                String(value),
+                                                                19
```
Comment (666lcz) [inline explorer/client/src/components/ownedobjects/OwnedObjects.tsx:380]: ditto

File: explorer/client/src/components/pagination/PaginationWrapper.tsx [hunk 3]
```diff
@@ -2,19 +2,26 @@
...
 import { useState, useCallback } from 'react';
 
+import TabFooter from '../../components/tabs/TabFooter';
+
 import styles from './PaginationWrapper.module.css';
 
 export default function PaginationWrapper({
     results,
     viewComponentFn,
+    ITEMS_PER_PAGE,
```
Comment (666lcz) [inline explorer/client/src/components/pagination/PaginationWrapper.tsx:12]:
```suggestion
    itemsPerPage,
```

nit: This is not a constant

File: explorer/client/src/components/pagination/PaginationWrapper.tsx [hunk 4]
```diff
@@ -35,86 +45,162 @@ export default function PaginationWrapper({
...
+                            )}
 
-            {viewComponentFn(objectSample)}
+                            {pageIndex > 1 && pageIndex < FINAL_PAGE_NO - 2 && (
+                                <button className={styles.pagenumber}>
+                                    {pageIndex + 1}
+                                </button>
+                            )}
+
+                            {pageIndex < FINAL_PAGE_NO - 3 && (
```
Comment (666lcz) [inline explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: CleanShot 2022-07-19 at 07 23 49

It looks like there's still a bug where `...` appears between 2850 and 2851

File: explorer/client/src/components/transactions-for-id/TxForID.tsx [hunk 5]
```diff
@@ -182,7 +188,13 @@ function TxForIDAPI({ id, category }: { id: string; category: categoryType }) {
     if (showData.loadState === 'loaded') {
         const data = showData.data;
         if (!data) return <></>;
-        return <PaginationWrapper results={data} viewComponentFn={viewFn} />;
+        return (
+            <PaginationWrapper
+                results={data}
+                viewComponentFn={viewFn}
+                ITEMS_PER_PAGE={10}
```
Comment (666lcz) [inline explorer/client/src/components/transactions-for-id/TxForID.tsx:195]: ditto

**Review by 666lcz — CHANGES_REQUESTED** [review]
Overall looks good, thanks for improving the pagination footer! Just a few code styling/naming nits and a bug in the pagination button logic

File: explorer/client/src/components/pagination/PaginationWrapper.tsx (hunk 4 above)
Comment (Jibz1) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: @666lcz that is a different component. I have a few modifications based on the feedback from Dreamten.

Comment (666lcz) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: I see. But is there any reason that we don't consolidate them and use one component for both?

Comment (Jibz1) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: I think we can

Comment (666lcz) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: @Andrew47 can we use the same component for both?

I also notice that there's some overflow if the page number is large in the component you are using

http://localhost:3000/addresses/0x813ec1ab1e1797c79d1e1fcbebebc27a1fd21f07

CleanShot 2022-07-19 at 07 35 53

Comment (mystie711) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: Here's the pagination conversation thread from Slack for quick reference,
https://mysten-labs.slack.com/archives/C03K869B8M7/p1658158419378789

Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: Where were you able to get example numbers in the thousands?

Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: Ah - I see. This is the Recent Transactions table in the Home Page. This PR focusses on the Address Results Page (see above description). I can handle this in a separate PR or we could get @Jibz1 to take a look given he has been recently working on the Transaction.

Comment (Jibz1) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx:163]: @Andrew47 I think we should combine the pagination component. The component on the home page already addresses all this, we just need to include an emit method that fires a function on the parent component. I think you already do this.

File: explorer/client/src/components/ownedobjects/OwnedObjects.tsx [hunk 6]
```diff
@@ -177,20 +181,33 @@ function OwnedObjectLayout({ results }: { results: resultType }) {
...
             {other_results.length > 0 && (
                 <div id="NFTSection">
-                    <h2>NFTs</h2>
+                    <div className={styles.ownedobjectheader}>
+                        <h2>NFTs</h2>
+                    </div>
                     <PaginationWrapper
                         results={other_results}
                         viewComponentFn={viewFn}
+                        ITEMS_PER_PAGE={6}
```
[the remark this answers is not in the document]
Comment (Andrew47) [reply explorer/client/src/components/ownedobjects/OwnedObjects.tsx]: Done

File: explorer/client/src/components/ownedobjects/OwnedObjects.tsx (hunk 1 above)
Comment (Andrew47) [reply explorer/client/src/components/ownedobjects/OwnedObjects.tsx]: Done

File: explorer/client/src/components/pagination/PaginationWrapper.tsx (hunk 3 above)
Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx]: Done

File: explorer/client/src/components/pagination/PaginationWrapper.tsx [hunk 7]
```diff
@@ -2,19 +2,26 @@
...
+    stats?: {
+        stats_text: string;
+        count: number;
+    };
 }) {
     const [pageIndex, setPageIndex] = useState(0);
 
-    const ITEMS_PER_PAGE = 12;
-
     const FINAL_PAGE_NO =
```
[the remark this answers is not in the document]
Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx]: Done

File: explorer/client/src/components/pagination/PaginationWrapper.tsx (hunk 4 above)
Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx]: I will merge the two components following the design specified here: https://www.figma.com/file/OzLaRFzevjxdQAbybWEZk0/Sui-Visualize?node-id=2979%3A59684. This is slightly different from what is both in the Home Page and in this PR.

**Review by 666lcz — APPROVED** [review]

File: explorer/client/src/components/pagination/PaginationWrapper.tsx (hunk 4 above)
Comment (Andrew47) [reply explorer/client/src/components/pagination/PaginationWrapper.tsx]: This is covered in the following PR: https://github.com/MystenLabs/sui/pull/3333

**Comment by Jibz1** [comment]
Screen Shot 2022-07-20 at 2 23 50 PM
@Andrew47

**Comment by Andrew47** [comment]
> Screen Shot 2022-07-20 at 2 23 50 PM @Andrew47

Well spotted - see PR here for a fix: https://github.com/MystenLabs/sui/pull/3418

**Pull request merged**